src/Core/Helpers/PathHelper.cs

```diff
@@ -149,5 +151,38 @@ private static string ToFilePathString(Uri uri)
             string escapedPath = uri.OriginalString.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
             return Uri.UnescapeDataString(escapedPath);
         }
+
+
+        /// <summary>
+        /// Gets Temp Directory under SLVS folder under %Temp%
+        /// </summary>
+        /// <param name="perVSInstance">Determines if the path will be per VS Instance</param>
+        /// <param name="paths">Creates sub folder structure under SLVS folder</param>
+        public static string GetTempDirForTask(bool perVSInstance, params string[] paths)
+        {
+            var SLVSTempFolder = Path.Combine(Path.GetTempPath(), "SLVS");
+
+            var taskPaths = new List<string> { SLVSTempFolder };
+
+            if (paths != null)
```

Comment:
`params` arguments are never null. If you don't specify any values the function is passed a zero-length array.

Reply:
It can be passed as null explicitly, however unlikely it will be. We can also add an assert for that scenario, but I think we should keep the check.